### PR TITLE
upgrade the vm service protocol library

### PR DESCRIPTION
**Pull request by devoncarew**
- upgrade the vm service protocol library

This brings in the latest version of the (generated) vm service protocol library, and adapts our debugger code for the relatively minor changes.

Note that the new version of the vm service protocol library is now forward compatible with major version changes of the protocol; we'll try and work with any version of the protocol.

### CODE REVIEW

File: src/io/flutter/vmService/VmServiceWrapper.java [hunk 1]
```diff
@@ -421,19 +448,64 @@ void received(List<Breakpoint> breakpointResponses, List<RPCError> errorResponse
...
+
+      }
+    }));
   }
 
   public void resumeIsolate(@NotNull final String isolateId, @Nullable final StepOption stepOption) {
     addRequest(() -> {
       myLatestStep = stepOption;
-      myVmService.resume(isolateId, stepOption, null, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
+      myVmService.resume(isolateId, stepOption, null, new ResumeConsumer() {
```
Comment (jacob314) [inline src/io/flutter/vmService/VmServiceWrapper.java:472]: can we update EMPTY_SUCCESS_CONSUMER or provide a new empty one for this case and the one starting on line 493?

File: third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java [hunk 2]
```diff
@@ -268,8 +258,9 @@ public void onError(RPCError error) {
       @Override
       public void received(Obj response) {
         if (response instanceof Instance) {
-          consumer.received((Instance) response);
-        } else {
+          consumer.received((Instance)response);
+        }
```
Comment (jacob314) [inline third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java:262]: nit: seems like surprising formatting. has the formatter been run?

**Review by jacob314 — APPROVED** [review]
lgtm

File: third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java (hunk 2 above)
Comment (devoncarew) [reply third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java:262]: This is straight from the generated code.

**Comment by helin24** [comment]
`JsonParser.parseString` seems to be causing issues in the kokoro process: https://sponge.corp.google.com/target?id=dd4f6b71-7843-44fd-842b-f3d2ab8ced45&target=flutter-intellij-kokoro%2Fmacos_external%2Fcontinuous&searchFor=

Is it maybe not available for Android Studio?

**Comment by devoncarew** [comment]
Shoot, that's likely a newer API. We can just change to using `new JsonParser().parse(...)`. Thanks for the heads up.

**Comment by devoncarew** [comment]
https://github.com/flutter/flutter-intellij/pull/4816

File: third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java (hunk 2 above)
Comment (devoncarew) [reply third_party/vmServiceDrivers/org/dartlang/vm/service/VmServiceBase.java:262]: Re-generation does correct the formatting here, so I might have unintentionally formatted this -

**Pull request merged**